File: solutions/typescript/difference-of-squares/1/difference-of-squares.ts
export class Squares {
  private n: number;

  constructor(count: number) {
    this.n = count;
  }

  get sumOfSquares(): number {
    return Array.from({ length: this.n }, (_, i) => (i + 1) * (i + 1))
      .reduce((acc, val) => acc + val, 0);
  }

  get squareOfSum(): number {
    const sum = Array.from({ length: this.n }, (_, i) => i + 1)
      .reduce((acc, val) => acc + val, 0);
    return sum * sum;
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/ruby/difference-of-squares/1/difference_of_squares.rb
class Squares
  def initialize(n)
    @n = n
  end

  def square_of_sum
    # Fórmula: (n * (n + 1) / 2) ^ 2
    sum = (@n * (@n + 1)) / 2
    sum**2
  end

  def sum_of_squares
    # Fórmula: n * (n + 1) * (2n + 1) / 6
    (@n * (@n + 1) * (2 * @n + 1)) / 6
  end

  def difference
    square_of_sum - sum_of_squares
  end
end